src/Table/Table.tsx

```diff
@@ -83,48 +75,60 @@ const Actions = styled(Td)(({ theme }) => ({
   },
 }))
 
+const InlineActionMenu = styled(ActionMenu)`
+  display: inline-flex;
+`
+
 const EmptyView = styled(Td)(({ theme }) => ({
   color: theme.color.text.default,
   height: 50,
   lineHeight: "50px",
   textAlign: "center",
 }))
 
-const Table: React.SFC<TableProps> = ({
-  rows,
-  columns,
-  onRowClick,
-  rowActionName,
-  __experimentalRowActions,
-  ...props
-}) => {
+function Table<T>({ data, columns, onRowClick, rowActions, ...props }: TableProps<T>) {
+  const standardizedColumns: Array<Column<T>> =
+    Boolean(columns[0]) && typeof columns[0] === "string"
+      ? (columns as Array<Extract<keyof T, string>>).map(columnName => ({
+          heading: columnName,
+          cell: (datum: T) => datum[columnName],
+        }))
+      : (columns as Array<Column<T>>)
   return (
     <Container {...props}>
       <thead>
         <Tr>
-          {columns.map((title, i) => (
-            <Th key={i}>{title}</Th>
+          {standardizedColumns.map((column, i) => (
+            <Th key={i}>{column.heading}</Th>
           ))}
-          {Boolean(__experimentalRowActions || rowActionName) && <Th />}
+          {Boolean(rowActions) && <Th />}
         </Tr>
       </thead>
       <tbody>
-        {rows.length ? (
-          rows.map((row, i) => (
+        {data.length ? (
+          data.map((datum, i) => (
```

Comment:
Can we be more explicit about the intent of `i` in this API?

```suggestion
          data.map((row, index) => (
```